src/core/expression/predicate.visitor.spec.js

```diff
@@ -0,0 +1,384 @@
+import {PredicateVisitor} from './predicate.visitor';
+
+function valueFactory() {
+	return (value) => value;
+}
+describe('PredicateVisitor', () => {
+
+	describe('visitGroup', () => {
+		it('check for equality with op = "and"', () => {
+			let group = {
+				kind: 'group',
+				op: 'and',
+				right: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				},
+				left: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				}
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visitGroup(group);
```

Comment:
Also, there's a minor error in typing for `expression.visitor`. `visit(item: IRoot, depth: number): void;` - depth is optional here.